### PR TITLE
Allow external library specification in modules.json

### PR DESCRIPTION
**Pull request by galpeter**
Added a way to specify external shared/static libraries to
be linked when a given module is enabled.

IoT.js-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by galpeter** [comment]
As mentioned in the #1362.

File: cmake/iotjs.cmake [hunk 1]
```diff
@@ -225,8 +232,15 @@ foreach(MODULE ${IOTJS_ENABLED_MODULES})
             message(FATAL_ERROR "C file doesn't exist: ${MODULE_PLATFORM_FILE}")
           endif()
         endforeach()
-      # ...otherwise add native files from 'undefined' section.
+
+        # external libraries....
+        foreach (idx ${${PLATFORMS_PREFIX}${IOTJS_SYSTEM_OS}.external_libs})
```
Comment (LaszloLango) [inline cmake/iotjs.cmake]: Please remove space after `foreach`

File: cmake/iotjs.cmake [hunk 2]
```diff
@@ -238,6 +252,12 @@ foreach(MODULE ${IOTJS_ENABLED_MODULES})
             message(FATAL_ERROR "${MODULE_UNDEFINED_FILE} does not exists.")
           endif()
         endforeach()
+
+        # external libraries....
+        foreach (idx ${${PLATFORMS_PREFIX}undefined.external_libs})
```
Comment (LaszloLango) [inline cmake/iotjs.cmake]: ditto

File: docs/devs/Writing-New-Module.md [hunk 3]
```diff
@@ -189,6 +189,27 @@ modules.json:
...
+Such libraries can be specified in the `modules.json` file so they will be linked when the IoT.js module is used.
+To specify third-party libraries the `external_libs` key should be used in the module specification.
+
+For example in the `modules.json`:
+```json
+{
+  "modules": {
+    "mymodule": {
+      "native_files": ["my_module.c"],
+      "external_libs": ["curl"],
```
Comment (daeyeon) [inline docs/devs/Writing-New-Module.md]: We seemingly need to allow writing this under `platforms`.
```
...
"platforms": {
   "linux": {
          "native_files": ["linux/my_module_platform_impl.c"],
          "external_libs": ["curl"]
   },
....
}
"native_files": ["my_module.c"],
"external_libs": ["lib_shared_on_all_platforms_if_it_truly_exists"],
...
```

**Comment by yichoi** [comment]
LGTM

**Pull request merged**